zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -663,6 +677,22 @@ impl Chain {
             .map(|(_height, tree)| tree.clone())
     }
 
+    /// Returns the Sapling [`NoteCommitmentSubtree`](sapling::tree::NoteCommitmentSubtree) specified
+    /// by a [`HashOrHeight`], if it exists in the non-finalized [`Chain`].
+    pub fn sapling_subtree(
+        &self,
+        hash_or_height: HashOrHeight,
+    ) -> Option<Arc<sapling::tree::NoteCommitmentSubtree>> {
+        let height =
+            hash_or_height.height_or_else(|hash| self.height_by_hash.get(&hash).cloned())?;
+
+        self.sapling_subtree
+            .iter()
+            .take(1)
+            .find(|subtree| subtree.end == height)
```

Comment:
Are we guaranteed that there will only ever be one subtree in the non-finalized state?
If we are, please document or assert that guarantee here, and fix the orchard code to match.

If not, let's remove `take(1)`:
```suggestion
        self.sapling_subtree
            .iter()
            .find(|subtree| subtree.end == height)
```

Reply:
Thank you.

> Are we guaranteed that there will only ever be one subtree in the non-finalized state?

No, I think there could be up to 5. It was originally only for finalizing blocks (which should have been documented).